### PR TITLE
Bump tokio, actix, reqwest

### PR DESCRIPTION
**Pull request by kelvinfan001**
Bump tokio to 1.5, actix to 0.11, reqwest to 0.11 and massage some APIs.

Closes https://github.com/coreos/zincati/issues/518

### CODE REVIEW

File: src/cli/agent.rs [hunk 1]
```diff
@@ -37,23 +37,24 @@ pub(crate) fn run_agent() -> Result<()> {
...
     trace!("creating actor system");
-    let sys = actix::System::builder()
-        .name(crate_name!())
-        .stop_on_panic(true)
-        .build();
+    let sys = actix::System::new();
 
-    trace!("creating metrics service");
-    let _metrics_addr = metrics::MetricsService::bind_socket()?.start();
+    sys.block_on(async {
```
Comment (kelvinfan001) [inline src/cli/agent.rs:42]: Self-note: API massaging here mostly caused by breaking changes caused by https://github.com/actix/actix-net/pull/207 in `actix-rt`.

**Comment by kelvinfan001** [comment]
Looks like the metrics service isn't started properly on the Unix-domain socket...

File: src/cli/agent.rs (hunk 1 above)
Comment (lucab) [reply src/cli/agent.rs:42]: I'm not sure if that is the cause of further issues, but all the address variables that were previously scoped to `main` are now only alive inside this `async` block.

Comment (kelvinfan001) [reply src/cli/agent.rs:42]: That should be fine right, since they're moved into structs later.

File: src/metrics/mod.rs [hunk 2]
```diff
@@ -51,12 +51,15 @@ impl Actor for MetricsService {
             .listener
             .try_clone()
             .expect("failed to clone metrics listener");
+        listener
+            .set_nonblocking(true)
```
Comment (kelvinfan001) [inline src/metrics/mod.rs:55]: This addresses https://github.com/coreos/zincati/pull/536#issuecomment-825335567.
In `tokio 0.3.2`, the `tokio::net::UnixListener::from_std()` method no longer helps us set Unix listeners to nonblocking, so we need to do it ourselves now.

**Comment by kelvinfan001** [comment]
@lucab This is ready for another look now :)

**Pull request merged**